### ISSUE TITLE
MONGO_URL not being implemented

### ISSUE THREAD

**Issue by tweakitguy** [arunoda/meteor-up-legacy#941]
Using MUP and attempting to connect to a mongodb via compose.io

For some reason the app is constantly connecting to my local mongoDB instead of considering the MONGO_URL.

Would this be do to an invalid mongo_url?  How do I confirm that my url is correct?

```
"MONGO_URL": "mongodb://user:pass@aws-us-east-1-portal.12.dblayer.com:10564/dbname",
 "MONGO_OPLOG_URL": "mongodb://oploguser:pass@haproxy627.aws-us-east-1-portal.12.dblayer.com:10627/local?authSource=dbname"
```

**Comment by MasterJames** [comment]
There is also the development branch mupx and a port of that version to kadirahq is still underway.
The url looks good but maybe the network is blocking traffic to those ports.
You can SSH into the target machine and see if you can telnet or similar to those ports. That or just check with the provider's ingress interface that the port is accessible.

**Comment by tweakitguy** [comment]
Had issues with mupx constantly failing for some reason.  MUP is working on this instance, and would prefer not to rebuild the server if possible.

I did have issues with having mongod 2.6 on the server, once updating to 3.2 I could access vis ssh but still my meteor app is choosing to use the local db after **mup reconfig**

Have also tried to redo **mup setup** and **mup deploy**. 

```
{
  // Server authentication info
  "servers": [
    {
       XXX
    }
  ],

  // Install MongoDB in the server, does not destroy local MongoDB on future setup
  "setupMongo": false,

  // WARNING: Node.js is required! Only skip if you already have Node.js installed on server.
  "setupNode": true,

  // WARNING: If nodeVersion omitted will setup 0.10.36 by default. Do not use v, only version number.
  "nodeVersion": "0.10.42",

  // Install PhantomJS in the server
  "setupPhantom": true,

  // Show a progress bar during the upload of the bundle to the server. 
  // Might cause an error in some rare cases if set to true, for instance in Shippable CI
  "enableUploadProgressBar": true,

  // Application name (No spaces)
  "appName": "XXX",

  // Location of app (local directory)
  "app": ".",

  // Configure environment
  "env": {
    "ROOT_URL": "http://XXX",
    "PORT": 3000,
    "MONGO_URL": "mongodb://XXXXX@aws-us-east-1-portal.12.dblayer.com:10564/XXXX",
    "MONGO_OPLOG_URL": "mongodb://XXXXX@haproxy627.aws-us-east-1-portal.12.dblayer.com:10627/local?authSource=lXXXX"
  },

  // Meteor Up checks if the app comes online just after the deployment
  // before mup checks that, it will wait for no. of seconds configured below
  "deployCheckWaitTime": 15
}
```

**Comment by jcharvet** [comment]
Can we access the external db now, I have been trying to put MONGO_URL everywhere without any success. my db is mLab I use also th eoplog no chance so far.

**Comment by MasterJames** [comment]
It should work? ! Did you get the prefix right mongodb://? Is the port number correct and not blocked from the target machine. SSH in to check it's connectable. Otherwise if you're not aware there is a port of mupx to kadirahq that might be working well enough to consider. It's still being worked on as this version is pretty much depreciated now but although it is technically stable it lacks many great additions and support moving forward.

**Comment by jcharvet** [comment]
Yes it worked, by looking at the source i figured it out, that I shouldn't setup the local mongo db then it would pickup my env variables.
Answer was in start.sh

**Comment by tweakitguy** [comment]
Still having issues here.  I can run 
MONGO_URL="mongodb://username:PASSWORD@hatch.mongohq.com:27017/test" meteor
and it connects to the remote DB, but using MUP always goes to server mongodb instead of using MONGO_URL

**Comment by jcharvet** [comment]
Did you say to the mupx setup to not install mongodb ?

**Comment by wdhif** [comment]
For me the MONGO_URL is working :

```
{
  "servers": [
    {
      "host": "XX",
      "username": "XX",
      "password": "XX"
    }
  ],
  "setupMongo": false,
  "setupNode": true,
  "nodeVersion": "0.10.43",
  "enableUploadProgressBar": true,
  "appName": "XXX",
  "app": ".",
  "env": {
    "PORT": 80,
    "ROOT_URL": "http://XXX.fr",
    "MONGO_URL": "mongodb://XXX:XXX@XX.XX.XX.XX:27017/XXX"
  },
  "deployCheckWaitTime": 15
}
```

With this configuration, the website is correctly using the external mongo base, (the mongodb is at digitalocean)

**Comment by tweakitguy** [comment]
Sorry, haven't had a chance to try.  Given up for the moment.  I will be creating a new server image and trying it out again.  Appreciate all the feedback.

**Comment by luisreyes** [comment]
I kept trying to connect to my mLab Sandbox DB not noticing that my `"setupMongo"` was set to true.
After setting this value to false it connected properly.
